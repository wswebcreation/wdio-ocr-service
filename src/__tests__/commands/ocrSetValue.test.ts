import ocrSetValue from '../../commands/ocrSetValue'
import ocrKeys from '../../utils/ocrKeys'
import ocrClickOnText from '../../commands/ocrClickOnText'

jest.mock('../../utils/ocrKeys', ()=> jest.fn())
jest.mock('../../commands/ocrClickOnText', ()=> jest.fn())

describe('ocrSetValue', () => {
  it('should be able to set a value', async () => {
    const options = {
      isTesseractAvailable: true,
      ocrImagesPath: 'ocrImagesPath',
      language: 'eng',
      reuseOcr: true,
      screenSize: { width: 1, height: 2 },
      text: 'text',
      value: 'value'
    }
    const globalAny:any = global
    globalAny.driver = {
      hideKeyboard: jest.fn(),
      waitUntil: jest.fn().mockImplementation((fn)=>fn()),
      isKeyboardShown: jest.fn(),
    }

    await ocrSetValue(options)

    expect(await ocrClickOnText).toHaveBeenCalledWithSnapshot()
    expect(await globalAny.driver.waitUntil).toHaveBeenCalledTimes(2)
    expect(await globalAny.driver.isKeyboardShown).toHaveBeenCalledTimes(2)
    expect(await ocrKeys).toHaveBeenCalledWith(options.value)
    expect(await globalAny.driver.hideKeyboard).toHaveBeenCalledTimes(1)
  })
})
