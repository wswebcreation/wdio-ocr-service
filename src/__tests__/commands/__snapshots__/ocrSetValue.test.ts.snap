// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ocrSetValue should be able to set a value 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "androidRectangles": undefined,
        "clickDuration": undefined,
        "iOSRectangles": undefined,
        "isTesseractAvailable": true,
        "language": "eng",
        "ocrImagesPath": "ocrImagesPath",
        "reuseOcr": true,
        "screenSize": Object {
          "height": 2,
          "width": 1,
        },
        "text": "text",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
